### PR TITLE
Updates to prepare for CBL-Mariner release

### PR DESCRIPTION
**Pull request by mamckee**
- Removed unnecessary build files from this repo
- Updated package install to place binaries and config files in the appropriate directory, rather than all in /usr/lib/keysinuse
- Added logic in engine bind to read config values directly, rather than rely on a .include directive in the default config

### CODE REVIEW

File: containers/Dockerfile-Alpine_Musl [hunk 1]
```diff
@@ -23,8 +23,8 @@ WORKDIR /keysinuse-openssl/packaging
 RUN make /keysinuse-openssl/bin/keysinuseutil
 
 # Copy files to staging
-RUN mkdir /usr/lib/keysinuse
-RUN cp /keysinuse-openssl/bin/keysinuse.so            /usr/lib/keysinuse
-RUN cp /keysinuse-openssl/bin/keysinuseutil           /usr/lib/keysinuse
-RUN cp /keysinuse-openssl/packaging/common/postinst    /usr/lib/keysinuse/installkeysinuse
-RUN chmod 0755 /usr/lib/keysinuse/installkeysinuse
\ No newline at end of file
+RUN mkdir /keysinuse
```
Comment (mlindgren) [inline containers/Dockerfile-Alpine_Musl:26]: Why in the root directory? /usr/lib might not be the right place, but I don't know if the root directory is either. Same for other dockerfiles.

File: engine/src/keysinuse_engine.c [hunk 2]
```diff
@@ -74,11 +77,67 @@ static int control(ENGINE *e, int cmd, long i, void *p, void (*func)(void))
...
+    // is disabled.
+    CONF *conf = NCONF_new(NULL);
+    conf = NCONF_new(NULL);
+    if (!NCONF_load(conf, keysinuse_conf_location, NULL))
+    {
+        log_error("Failed to load keysinuse config,OPENSSL_%ld", ERR_get_error());
+        goto err;
+    }
+
+    int i, success;
```
Comment (mlindgren) [inline engine/src/keysinuse_engine.c]: >int i, [](http://example.com/codeflow?start=4&length=6)

Nit: define this in the loop? #Resolved

Comment (mlindgren) [inline engine/src/keysinuse_engine.c]: >success [](http://example.com/codeflow?start=11&length=7)

Nit: initialize #Resolved

File: engine/src/keysinuse_engine.c [hunk 3]
```diff
@@ -74,11 +77,67 @@ static int control(ENGINE *e, int cmd, long i, void *p, void (*func)(void))
...
+    long num_val = 0;
+    char *string_val = NULL;
+    for (i = 0; i < (sizeof(supported_cmds) / sizeof(ENGINE_CMD_DEFN)) && supported_cmds[i].cmd_num != 0; i++)
+    {
+        if (supported_cmds[i].cmd_flags & ENGINE_CMD_FLAG_STRING)
+        {
+            string_val = NCONF_get_string(conf, keysinuse_conf_section, supported_cmds[i].cmd_name);
```
Comment (mlindgren) [inline engine/src/keysinuse_engine.c:105]: >string_val [](http://example.com/codeflow?start=12&length=10)

Can't find documentation for NCONF_get_string - I'm guessing it returns a pointer into a buffer owned by the CONF object, so this string doesn't need to be freed separately? #Resolved

File: packaging/common/postinst [hunk 4]
```diff
@@ -23,9 +23,15 @@ case $1 in
...
+  if [ -x /usr/bin/keysinuseutil ]; then
     echo "Enabling keysinuse engine for OpenSSL"
-    /usr/lib/keysinuse/keysinuseutil -update-default install|| echo "Configuring engine failed"
+    /usr/bin/keysinuseutil -update-default install || echo "Configuring engine failed"
+  fi
+
+  # Config used to reside in /usr/lib/keysinuse. Create symlink for
+  # backwards compat and discoverability
+  if [ -e /etc/keysinuse/keysinuse.cnf ]; then
+    ln -s /etc/keysinuse/keysinuse.cnf /usr/lib/keysinuse/keysinuse.cnf
```
Comment (mlindgren) [inline packaging/common/postinst:34]: >ln -s /etc/keysinuse/keysinuse.cnf /usr/lib/keysinuse/keysinuse.cnf [](http://example.com/codeflow?start=4&length=67)

What's the scenario we're concerned about here? Someone on a machine using the old version upgrades to the new version? In that case you're just overwriting their existing config file with a symlink. I don't see the benefit here. #Resolved

File: packaging/common/preinst [hunk 5]
```diff
@@ -3,9 +3,9 @@
 case $1 in
   install | upgrade | 1 | 2)
   # Check if the engine is already configured. Deconfigure first if so
-  if [ -x /usr/lib/keysinuse/keysinuseutil ]; then
+  if [ -x /usr/bin/keysinuseutil ]; then
```
Comment (mlindgren) [inline packaging/common/preinst:6]: >if [ -x /usr/bin/keysinuseutil ]; then [](http://example.com/codeflow?start=2&length=38)

Should you also check for the existence of the old version in the old location?

File: packaging/common/prerm [hunk 6]
```diff
@@ -2,16 +2,22 @@
 
 case $1 in
   remove | upgrade | 0)
-  # Deconfigure engine
-  if [ -x /usr/lib/keysinuse/keysinuseutil ]; then
+  # Deconfigure enginepo
```
Comment (mlindgren) [inline packaging/common/prerm]: >enginepo [](http://example.com/codeflow?start=16&length=8)

enginepo? #Resolved

**Comment by mlindgren** [comment]
> #!/bin/bash

Should this file still be in this repo?

---
In reply to: [1155807468](https://github.com/microsoft/KeysInUse-OpenSSL/pull/1#issuecomment-1155807468) [](http://example.com/codeflow?ancestors=1155807468)

---
Refers to: scripts/version.sh:1 in ec36cd0. [](commit_id = ec36cd0372ccbd526675a00f22726b4d01ce5a73, deletion_comment = False)

File: packaging/common/postinst (hunk 4 above)
Comment (mamckee) [reply packaging/common/postinst:34]: This is just to give us a temporary buffer as we upgrade existing deployments. Our guidance and deployment model has been to use '.include' in the default OpenSSL config, and point to this location. We should be safe to remove this, but I'd like to keep this as a failsafe when we upgrade and then remove in 0.3.2

File: engine/src/keysinuse_engine.c (hunk 3 above)
Comment (mamckee) [reply engine/src/keysinuse_engine.c:105]: There isn't any docs for 1.1.1 on this API as far as I can tell, but that's correct. The entire config gets freed all at once.

**Comment by mamckee** [comment]
> > #!/bin/bash
> 
> Should this file still be in this repo?
> 
> Refers to: scripts/version.sh:1 in [ec36cd0](https://github.com/microsoft/KeysInUse-OpenSSL/commit/ec36cd0372ccbd526675a00f22726b4d01ce5a73). [](commit_id = [ec36cd0](https://github.com/microsoft/KeysInUse-OpenSSL/commit/ec36cd0372ccbd526675a00f22726b4d01ce5a73), deletion_comment = False)

No, it shouldn't

File: packaging/rpm/keysinuse.spec [hunk 7]
```diff
@@ -6,7 +6,7 @@ Release:     LABEL
 Summary:     KeysInUse solution for OpenSSL
 
 Group:       Productivity/Security
-License:     Apache
+License:     MIT
 URL:         www.microsoft.com
```
Comment (samuel-lee-msft) [inline packaging/rpm/keysinuse.spec]: I think this URL should point back to this git repo: `https://github.com/microsoft/KeysInUse-OpenSSL`

File: packaging/rpm/keysinuse.spec [hunk 8]
```diff
@@ -35,7 +35,7 @@ rm -rf $RPM_BUILD_ROOT
 %files
 %defattr(-,root,root,-)
```
Comment (samuel-lee-msft) [inline packaging/rpm/keysinuse.spec:36]: May want to include `%license LICENSE`

**Review by mlindgren — APPROVED** [review]
:shipit:

**Pull request merged**